### DISCUSSION TITLE
NamedIndexes and ReadingContext Replacement?

### DISCUSSION THREAD

**Discussion by dprice** [JoshClose/CsvHelper#1765] [Q&A] [answered]
After upgrading from v12 to v26, I lost the ability to get detailed information about a failing CSV read. This is a sample of what I was doing before:

```
Reader.Configuration.ReadingExceptionOccurred = ex =>
{
    var fieldNames = ex.ReadingContext.NamedIndexes.Keys.ToList();
    var fieldName = fieldNames.Any() ? fieldNames[ex.ReadingContext.CurrentIndex] : "Unknown";
    var msg =
        $"Unable to convert field '{fieldName}' with value '{ex.ReadingContext.Record[ex.ReadingContext.CurrentIndex]}' on row {ex.ReadingContext.Row}. Actual data: {ex.ReadingContext.RawRecord}. ";
    msg += $"Additional Information: {ex}";
    _logger.Error(msg);
    
    return false;
};
```

I've updated this to use `Exception.Context` for `ReadingContext` but I cannot get the same level of detail that I had before. Where did these properties migrate to?

**Comment by JoshClose** [comment, marked as answer]
They're internal to classes. Having them globally available like that slowed reading down significantly.

Most of that information should appear in the exception message now though.
